rfd/0022-ssh-agent-forwarding.md

```diff
@@ -0,0 +1,224 @@
+        │─────────────────────>│                        │                         │
+        │                      │    Signed Response     │                         │
+        │                      │───────────────────────>│                         │
+        │              ╔═══════╧════════════════════════╧════════╗                │
+        │              ║       Authentication established        ║                │
+        │              ╚═══════╤════════════════════════╤════════╝                │
+        │                      │               ╔════════╧═════════╗               │
+        │                      │               ║ User connects to ║               │
+        │                      │               ║ "Third Machine"  ║               │
+        │                      │               ╚════════╤═════════╝               │
+        │                      │                        │       SSH Connect       │
+        │                      │                        │ ───────────────────────>│
+        │                      │                        │     Auth challenge      │
+        │                      │                        │ <───────────────────────│
+        │  Auth Challenge (via forwarded connection)    │                         │
+        │<──────────────────────────────────────────────│                         │
+        │  Signed Response (via forwarded connection)   │                         │
+        │──────────────────────────────────────────────>│                         │
+        │                      │                        │     Signed Response     │
+        │                      │                        │ ───────────────────────>│
+        │                      │                ╔═══════╧═════════════════════════╧═══════╗
+        │                      │                ║       Authentication established        ║
+        |                      |                ╚═════════════════════════════════════════╝
+   ┌────┴────┐          ┌──────┴──────┐          ┌──────┴───────┐          ┌──────┴──────┐
+   │Key Agent│          │Local Machine│          │Remote Machine│          │Third Machine│
+   └─────────┘          └─────────────┘          └──────────────┘          └─────────────┘
+```
+
+The above diagram leaves out a _lot_ of detail, but the overall process should be clear
+enough for this discussion.
+
+### What's the issue?
+
+An OpenSSH user can forward their Key Agent to the remote machine using the `-A` flag on
+the `ssh` command line. This allows the user to use arbitrary keys stored on their _local_
+machine to authenticate against a "Third Machine" from inside their session on the Remote
+Machine.
+
+When running `tsh ssh`, a user may have _two_ independent key agents running: their own
+(which we will call the _User Key Agent_), and an ephemeral one inside the `tsh` process
+itself (the _Teleport Key Agent_). The Teleport Key Agent is populated with the contents
+of the user's `.tsh` directory.
```

Comment:
Can you check what we call this in code, I think we refer to the agent at `SSH_AUTH_SOCK` as the system agent.

Reply:
Most commonly referred to as the _System Key Agent_. Have updated RFD to match.